### PR TITLE
feat: fetch audience id for retl criteo audience

### PR DESCRIPTION
**Pull request by ruchiramoitra**
## Description of the change
fetch audience id from context for retl source and criteo destination if mapped in vdm
> Description here

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

**Review by anantjain45823 — APPROVED** [review]

File: test/__tests__/data/criteo_audience_router_rETL_input.json [hunk 1]
```diff
@@ -0,0 +1,63 @@
+[
+  {
+    "destination": {
+      "Config": {
+        "clientId": "abcdef8-f49-4cd6-b4c5-958b3d66d431",
+        "clientSecret": "sjhdkhfrz6yc9LrRRIPimE9h53jADLccXTykHCcA6eEoFR4rXQg",
+        "audienceId": "34893",
```
Comment (krishna2020) [inline test/__tests__/data/criteo_audience_router_rETL_input.json:7]: Can you add one more test case where `audienceId` is null or not available as that's the current code change ?

File: src/v0/destinations/criteo_audience/transform.js [hunk 2]
```diff
@@ -37,19 +37,19 @@ const responseBuilder = (message, destination, accessToken) => {
   const { Config } = destination;
   const { audienceId } = Config;
   const { listData } = message.properties;
-  operation.forEach(op => {
+
+  let finalAudienceId = audienceId;
+  const mappedToDestination = get(message, MappedToDestinationKey);
+  if (!finalAudienceId && mappedToDestination) {
+    const { objectType } = getDestinationExternalIDInfoForRetl(message, 'CRITEO_AUDIENCE');
```
Comment (krishna2020) [inline src/v0/destinations/criteo_audience/transform.js:44]: As we didn't make the `audienceId` mandatory in the configuration flow, is this additional logic being required ?

Comment (ruchiramoitra) [reply src/v0/destinations/criteo_audience/transform.js:44]: here we are checking if audienceId is present in the config itself or in the context. 
As we already have vdm enabled and now we have audienceId in the dest config such existing connection will not have audienceid in the context. so this is for backward compatibilty. (though we have not announced criteo aud yet so very less chance of such cases)

File: test/__tests__/data/criteo_audience_router_rETL_input.json (hunk 1 above)
Comment (ruchiramoitra) [reply test/__tests__/data/criteo_audience_router_rETL_input.json:7]: added.

File: test/__tests__/data/criteo_audience_router_rETL_input.json [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+          "job_run_id": "cgiiurt8um7k7n5dq480",
+          "task_run_id": "cgiiurt8um7k7n5dq48g",
+          "job_id": "2MUWghI7u85n91dd1qzGyswpZan",
+          "version": "895/merge"
+        }
+      },
+      "timestamp": "2020-02-02T00:23:09.544Z"
+    }
+  },
+  {
```
Comment (krishna2020) [inline test/__tests__/data/criteo_audience_router_rETL_input.json:63]: there's no corresponding expected response for this test payload

Comment (ruchiramoitra) [reply test/__tests__/data/criteo_audience_router_rETL_input.json:63]: my mistake. forgot to git add the file

File: dist/src/v0/destinations/criteo_audience/transform.js
Comment (krishna2020) [inline dist/src/v0/destinations/criteo_audience/transform.js:1]: can you remove these distribution files ?

Comment (ruchiramoitra) [reply dist/src/v0/destinations/criteo_audience/transform.js:1]: done

Comment (krishna2020) [inline dist/src/v0/destinations/criteo_audience/transform.js:1]: delete this file

**Review by anantjain45823 — APPROVED** [review]
